### PR TITLE
Update requirements.txt

### PR DESCRIPTION
**Pull request by cyrilzakka**
Added required packages to run llama fine-tune/rlhf experiments.

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for the PR!
I don't think we will merge the PR as it is, as it depends on various forks, that they depend themselves on a specific commit. This can make the library quite hard to maintain. However, you can specify more instructions to users  on this PR that want to perform what you had in mind, so that they can easily reproduce.

**Comment by lvwerra** [comment]
Agree, we shouldn't add this to the library requirements but we could move it into either the readme of the StackLlama example or have a requirement there.

**Comment by cyrilzakka** [comment]
@younesbelkada @lvwerra  Whatever you think is best!  I was just trying to replicate the StackLLaMa experiments and the dependencies required didn't match the ones in the repo. Thought I'd save others some time!

**Comment by lvwerra** [comment]
If you move the requirements to the StackLlama folder I am happy to add it. But we shouldn't enforce it at the library level.

**Comment by cyrilzakka** [comment]
Done, and fixed a typo in the StackLlama ReadMe

**Review by lvwerra — COMMENTED** [review]
Sounds good. @younesbelkada could we already go from the official repos' `main` or is there even a release with those fixes?

**Review by lvwerra — APPROVED** [review]
Looks good!

**Review by younesbelkada — COMMENTED** [review]
Thanks for iterating, left one comment

File: examples/stack_llama/scripts/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+datasets>=1.17.0
+torch>=1.4.0
+tqdm
+accelerate
+trl 
+bitsandbytes==0.37.2 
+git+https://github.com/mymusise/peft.git@chatglm 
+git+https://github.com/zphang/transformers.git@llama_push
```
[the remark this answers is not in the document]
Comment (lvwerra) [reply examples/stack_llama/scripts/requirements.txt:8]: true, and same for peft?

**Pull request closed without merging**